edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleConnection.cs

```diff
@@ -86,6 +86,30 @@ async Task<IModuleClient> InitModuleClient()
             }
         }
 
+        async Task EnableSubscriptions(IModuleClient moduleClient)
+        {
+            try
+            {
+                await moduleClient.SetDefaultMethodHandlerAsync(this.MethodCallback);
+                await moduleClient.SetDesiredPropertyUpdateCallbackAsync(this.desiredPropertyUpdateCallback);
+            }
+            catch (Exception ex)
+            {
+                Events.ErrorSettingUpNewModule(ex);
+
+                try
+                {
+                    await moduleClient.CloseAsync();
+                }
+                catch
+                {
+                    // swallowing intentionally
+                }
+
+                throw;
```

Comment:
Where would this get caught/handled?

Reply:
the main idea of the rethrow was to not change the existing error handling, but
- this is called from an OnClosed() event handler, and it catches and swallows the exception
- it is called by 'GetTwinFromIotHub' which catches it and either retries or just swallows the exception (retries once)
- called from the ModuleClient ctor - there the idea I suppose that the module client subscribes to events (method handler and desired properties) plus when the connection gets closed, it reopens. I think this latter logic might be broken as I see, because started offline, the method will throw an exception and I don't see how it would automatically create a new connection anytime soon (it will when some periodic operation, like twin pull requests a new connection) - but again, with this change the logic stays the same. 

added a ticket for this last question: (10555951)